### PR TITLE
Add repo documentation

### PR DESCRIPTION
**Pull request by SpirinEgor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,172 @@
...
+```
+- Predict target sequence for given file with concrete model (predict function name by its body),
+this step includes data preprocessing and model applying:
+```bash
+python interactive.py test.java model.pt
+```
+
+## Evaluation
+
+Currently, there are implemented Tree-LSTM model with various modifications. For all of them, LSTM with Attention to subtrees
```
Comment (areyde) [inline README.md:110]: Currently, several Tree-LSTM models with various modifications are implemented. For all of them, LSTM with Attention to subtrees is used to decode the representation into a target sequence (a method name). As for the node embedding, the matrix embeddings for types (internal nodes) and tokens (leafs) are employed.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,172 @@
...
+is used to decoder to target sequence (method name). For node embedding used type (internal nodes) and token (leafs) matrix embeddings.
+
+<table>
```
Comment (areyde) [inline README.md:113]: I think this table requires some commentary. Still, what are the names on the left? The reader doesn't know.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,172 @@
...
+    <td>56.97</td>
+    <td>38.86</td>
+    <td>46.20</td>
+  </tr>
+</tbody>
+</table>
+
+## Contribution
+
+Supporting different algorithms to encode and decode trees are the main key areas of improvement this framework.
```
Comment (areyde) [inline README.md:171]: Supporting different algorithms of encoding and decoding trees is the key area of improvement for this framework.
If you have any suggestions or questions, feel free to open issues and create pull requests

File: README.md [hunk 4]
```diff
@@ -1,4 +1,172 @@
...
+    <td>450 260</td>
+  </tr>
+  <tr>
+    <td>Java-large</td>
+    <td><a href="https://s3-eu-west-1.amazonaws.com/datasets.ml.labs.aws.intellij.net/java-ast-methods/java-ast-methods-large.tar.gz">s3 link</a></td>
+    <td>8999</td>
+    <td>250</td>
+    <td>307</td>
+    <td>16 794 775</td>
+    <td>~350 000</td>
```
[the remark this answers is not in the document]
Comment (SpirinEgor) [reply README.md:68]: Yes, it seems very strange, that it's close to the medium dataset. But splitting into `train/test/val` came from code2seq paper, I just reuse it.

File: README.md (hunk 2 above)
Comment (SpirinEgor) [reply README.md:113]: Put the link to my thesis. I think there isn't much space to put all descriptions about modifications.

**Pull request merged**